src/napari_imagej/widget_IJ2.py

```diff
@@ -43,12 +44,82 @@ def __init__(self, viewer: Viewer):
             self.gui_button.clicked.connect(lambda: self.to_ij.setEnabled(True))
             self.gui_button.clicked.connect(lambda: self.from_ij.setEnabled(True))
 
+    @property
+    def gui(self) -> "jc.UserInterface":
+        """Convenience function for obtaining the default UserInterface"""
+        ensure_jvm_started()
+        return ij().ui().getDefaultUI()
+
     def _showUI(self):
         """
         NB: This must be its own function to prevent premature calling of ij()
         """
         ensure_jvm_started()
-        ij().ui().showUI()
+        # First time showing
+        if not self.gui.isVisible():
+            # First things first, show the GUI
+            ij().ui().showUI(self.gui)
+            # Then, add our custom settings to the User Interface
+            if ij().legacy and ij().legacy.isActive():
+                self._ij1_UI_setup()
+            else:
+                self._ij2_UI_setup()
+        # Later shows - the GUI is "visible", but the appFrame probably isn't
+        else:
+            self.gui.getApplicationFrame().setVisible(True)
```

Comment:
We should update the documentation to reflect the fact that closing the GUI no longer prevents issues.

Reply:
Done